packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/renderTeachingPopoverCarousel.tsx

```diff
@@ -20,17 +19,19 @@ export const renderTeachingPopoverCarousel_unstable = (
   const { layout } = state;
 
   return (
-    <TeachingPopoverCarouselProvider value={contextValues.carousel}>
-      <state.root>
-        {state.root.children}
-        <state.footer>
-          {layout === 'centered' && state.previous && <state.previous />}
-          {state.nav && <state.nav />}
-          {state.pageCount && <state.pageCount />}
-          {layout === 'offset' && state.previous && <state.previous />}
-          <state.next />
-        </state.footer>
-      </state.root>
-    </TeachingPopoverCarouselProvider>
+    <CarouselWalkerProvider value={contextValues.carouselWalker}>
+      <CarouselProvider value={contextValues.carousel}>
+        <state.root>
+          {state.root.children}
+          <state.footer>
```

Comment:
> ⚠️ This is not blocking - but worth considering either now or for the standalong Carousel.

Looking at the styling of the footer you could actually extract that to be a separate component. This would:

* Remove a nested slot level
* Make footer customizations much more straightforward

```tsx
TeachingPopoverBody value={'test-1'} media={<Image alt={'test image'} fit={'cover'} src={swapImage} />}>
  <TeachingPopoverTitle>{'Teaching Bubble Title'}</TeachingPopoverTitle>
  <div>{`This is page: 2`}</div>
  <TeachingPopoverCarouselFooter className={styles.override} next={{className:"styles.override"}} previous={{className="override"}} pageCount={{className: "override"}} />
</TeachingPopoverBody>
```